Add github reporter status_url config option

ConsoleCatzirl opened the pull request:
Make it possible to set the github status link to something other than the
zuul-server status url. This allows the github status to link to the log output
for the jobs. If set, this url will be run through the same formatter
as the url_pattern used for comment links. If unset, it will fall back to the
previous behaviour of using the zuul_server status_url.

Also make the github reporter report() function take the same argument list as
the abstract base class.

Change-Id: I5f2ff35bb38426ddcd7f65798b4f114256a54847

omgjlk commented on zuul/reporter/github.py:
```diff
@@ -75,11 +75,15 @@ def setPullStatus(self, pipeline, item):
+
         url = ''
-        if self.sched.config.has_option('zuul', 'status_url'):
+        if self._status_url:
```
This could use some code commentary around the logic used. It's somewhat difficult to follow.

ConsoleCatzirl replied:
Done

omgjlk reviewed (commented):
Is there an upstream change request to match this one?

ConsoleCatzirl committed:
Add github reporter status_url config option

Make it possible to set the github status link to something other than the
zuul-server status url. This allows the github status to link to the log output
for the jobs. If set, this url will be run through the same formatter
as the url_pattern used for comment links. If unset, it will fall back to the
previous behaviour of using the zuul_server status_url.

Also make the github reporter report() function take the same argument list as
the abstract base class.

Change-Id: I5f2ff35bb38426ddcd7f65798b4f114256a54847

ConsoleCatzirl commented:
I've proposed this upstream as https://review.openstack.org/415750

jamielennox commented:
I think you'll need to cherry pick your change onto the existing github branch, not merge the whole thing.

jamielennox commented:
recheck
